tests/libgstc/test_libgstc_debug.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * GStreamer Daemon - Gst Launch under steroids
+ * Copyright (c) 2015-2018 Ridgerun, LLC (http://www.ridgerun.com)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+#include <gst/check/gstcheck.h>
+#include <string.h>
+
+#include "libgstc.h"
+#include "libgstc_json.h"
+#include "libgstc_socket.h"
+
+/* Test Fixture */
+static gchar _request[3][512];
+static GstClient *_client;
+int request_count = 0;
+
+static void
+setup ()
+{
+  const gchar *address = "";
+  unsigned int port = 0;
+  unsigned long wait_time = 5;
+  int keep_connection_open = 0;
+
+  gstc_client_new (address, port, wait_time, keep_connection_open, &_client);
+}
+
+static void
+teardown (void)
+{
+  gstc_client_free (_client);
+}
+
+/* Mock implementation of a socket */
+typedef struct _GstcSocket
+{
+} GstcSocket;
+
+GstcSocket _socket;
+
+GstcStatus
+gstc_socket_new (const char *address, const unsigned int port,
+    const unsigned long wait_time, const int keep_connection_open,
+    GstcSocket ** out)
+{
+  *out = &_socket;
```

Comment:
Add pointer guards

Reply:
Is this necessary since these are only placeholder functions used only during these tests?

Reply:
It isn't necessary, but it's a good practice since another programmer could modify the test, send null values and don't realize why the tests are failing.